### DISCUSSION TITLE
Use `meta_data` field to covey an asset's `characteristic`

### DISCUSSION THREAD

**Discussion by dstadulis** [lightninglabs/taproot-assets#763] [Ideas]
### Background
Using integers in applications which require uncompromising accuracy (Money, accounting, settlement etc) is common in order to handle the onerous computer hardware and software demands to perform accurate operations on floating point numbers.  IEEE 754 is a standards example of the attempts to create uniformity in how to implement computation on floats.
Financial accounting software often uses integers to achieve accuracy without subjecting itself to FP arithmetic complexity.
https://en.wikipedia.org/wiki/IEEE_754
https://husobee.github.io/money/float/2016/09/23/never-use-floats-for-currency.html

tapd uses integers to ensure that it's precision and accuracy is high.  Bitcoin similarly uses integers for internal account and `COIN = 100000000` used to convert/present the decimal in the UI.

Such a variable merely instructs UIs on where to place the decimal point and referred to as a `characteristic`. [1]

### User / Market Needs
As a `application developer` I want an `programmatic, authenticated method to produce an asset's characteristic` in order to `make the presentation of tapd's internally-accounted integers be displayed in UIs without needing bespoke code`
As an `Asset Issuer` I want to `convey where the decimal place should be displayed to users` in order to `prevent confusion and calculation errors in handling amounts`

### Resources
[1] https://en.wikipedia.org/wiki/Characteristic_(exponent_notation)#Mantissa_and_characteristic
https://en.wikipedia.org/wiki/Significand

**Comment by RajGoodF** [comment]
Should we standardize this meta key, because as this is meta field every node/wallet/ui can interpret as per there requirements, so we should not allow this and create a standard protocol/rule which all nodes inherit while they sync asset and way they interpret this rule stays intact forever irrespective of node/ui.

**Reply by Roasbeef** [reply]
Yeah we should, can you see #925? We're working on some of the design here and would like feedback.
